File: week6/cash.py
from cs50 import get_float

def main():
    cash = get_cash()
    cents = round(cash * 100)  # 避免浮點誤差

    coins = 0

    for coin in [25, 10, 5, 1]:
        coins += cents // coin
        cents %= coin

    print(coins)

def get_cash():
    while True:
        n = get_float("Change owed: ")
        if n > 0:
            return n

main()

File: week1/cash/cash.c
#include<stdio.h>
#include<cs50.h>

int main(void)
{
    int cash;
    int quarters=0;
    int dimes =0;
    int nickels=0;
    int pennies=0;

    do
    {
        cash = get_int("Change owed: ");
    }
    while(cash < 0);

    for(int i = 0; i < cash/25; i++)
    {
        quarters+=1;
    }
    cash-= 25*quarters;
    for(int i = 0; i < cash/10; i++)
    {
        dimes+=1;
    }
    cash-= 10*dimes;
    for(int i = 0; i < cash/5; i++)
    {
        nickels+=1;
    }
    cash-= 5*nickels;
    for(int i = 0; i < cash/1; i++)
    {
        pennies+=1;
    }
    cash-= 1*pennies;
    printf("%i\n", pennies+nickels+dimes+quarters);
}
